python/lsst/pipe/tasks/selectImages.py

```diff
@@ -339,3 +339,125 @@ def runDataRef(self, dataRef, coordList, makeDataRefList=True, selectDataList=[]
+        """
+        if self.config.nImagesMax <= 0:
+            raise RuntimeError(f"nImagesMax must be greater than zero: {self.config.nImagesMax}")
+
+        psfSizes = []
+        dataRefList = []
+        exposureInfoList = []
+
+        patchVertices = [coord.getVector() for coord in coordList]
+        patchPoly = lsst.sphgeom.ConvexPolygon.convexHull(patchVertices)
+
+        for data in selectDataList:
+            dataRef = data.dataRef
+            imageWcs = data.wcs
+            cal = dataRef.get("calexp", immediate=True)
+            psfSize = cal.getPsf().computeShape().getDeterminantRadius()
+            nx, ny = cal.getDimensions()
+
+            imageBox = afwGeom.Box2D(afwGeom.Point2D(0, 0), afwGeom.Extent2D(nx, ny))
+            try:
+                imageCorners = [imageWcs.pixelToSky(pix) for pix in imageBox.getCorners()]
+            except (pexExceptions.DomainError, pexExceptions.RuntimeError) as e:
+                # Protecting ourselves from awful Wcs solutions in input images
+                self.log.debug("WCS error in testing calexp %s (%s): deselecting", dataRef.dataId, e)
+                continue
+
+            imagePoly = lsst.sphgeom.ConvexPolygon.convexHull([coord.getVector() for coord in imageCorners])
+            if imagePoly is None:
+                self.log.debug("Unable to create polygon from image %s: deselecting", dataRef.dataId)
+                continue
+
+            if patchPoly.intersects(imagePoly):
+                # "intersects" also covers "contains" or "is contained by",
+                # so there is no guarantee the whole area is covered
+
+                # if min/max PSF values are defined, remove images out of bounds
+                sizeFwhm = psfSize * np.sqrt(8.*np.log(2.))
+                if self.config.maxPsfFwhm and sizeFwhm > self.config.maxPsfFwhm:
+                    continue
+                if self.config.minPsfFwhm and sizeFwhm > self.config.minPsfFwhm:
+                    continue
+                psfSizes.append(psfSize)
+                dataRefList.append(dataRef)
+                exposureInfoList.append(BaseExposureInfo(dataRef.dataId, imageCorners))
+
+        if len(psfSizes) > self.config.nImagesMax:
+            sortedIndices = np.argsort(psfSizes)[:self.config.nImagesMax]
+            filteredDataRefList = [dataRefList[i] for i in sortedIndices]
+            filteredExposureInfoList = [exposureInfoList[i] for i in sortedIndices]
+            self.log.info(f"{len(sortedIndices)} images selected with FWHM "
+                          f"range of {psfSizes[sortedIndices[0]]}--{psfSizes[sortedIndices[-1]]} pixels")
+
+        else:
+            if len(psfSizes) == 0:
+                self.log.info(f"0 images selected.")
+            else:
+                self.log.info(f"{len(psfSizes)} images selected with FWHM range "
+                              f"of {psfSizes[0]}--{psfSizes[-1]} pixels")
```

Comment:
This should almost certainly be `debug` -- it's not information of use to somebody running the pipeline normally.